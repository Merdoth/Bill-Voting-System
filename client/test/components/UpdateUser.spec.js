import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { UpdateUser } from
  '../../js/components/UpdateUser';

window.$ = require('jquery');

/* global jest */

const props = {
  user: {
    token: {}
  },
  getAUser: () => { },
  updateUser: jest.fn(),
  logout: {}
};

const setup = () => shallow(<UpdateUser {...props} />);

describe('UpdateUser Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should return user', () => {
    const wrapper = setup();
    wrapper.setState({
      fullName: 'Samuel Uyali',
      userName: 'Samuel',
      email: 'samuel.uyali@gmail.com',
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onSubmit()', () => {
  it('should update user to the state', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const wrapper = setup();
    const form = wrapper.find('form');
    form.simulate('submit', event);
    expect(props.updateUser).toHaveBeenCalled();
  });
});

describe('componentWillReceiveProps()', () => {
  it('should call componentWillReceiveProps()', () => {
    const wrapper = setup();
    wrapper.setProps({ userFound: { fullName: 'test' } });
    expect(wrapper.state('fullName')).toEqual('test');
  });
});

describe('onChange()', () => {
  it('should set fullName to state when input values changes', () => {
    const event = {
      target: { name: 'fullName', value: '' }
    };
    const wrapper = setup();
    const fullNameInput = wrapper.find('#fullName');

    event.target.value = 'Sarah Gigs';
    fullNameInput.simulate('change', event);

    expect(wrapper.instance().state.fullName).toBe('Sarah Gigs');
  });

  it('should set userName state when input values changes', () => {
    const event = {
      target: { name: 'userName', value: '' }
    };
    const wrapper = setup();
    const userNameInput = wrapper.find('#userName');

    event.target.value = 'Sarah';
    userNameInput.simulate('change', event);

    expect(wrapper.instance().state.userName).toBe('Sarah');
  });

  it('should set email state when input values changes', () => {
    const event = {
      target: { name: 'email', value: '' }
    };
    const wrapper = setup();
    const emailInput = wrapper.find('#email');

    event.target.value = 'sarah.gigs@gmail.com';
    emailInput.simulate('change', event);

    expect(wrapper.instance().state.email).toBe('sarah.gigs@gmail.com');
  });
});
