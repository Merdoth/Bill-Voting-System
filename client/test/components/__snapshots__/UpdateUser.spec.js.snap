// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateUser Component snapshot it should render the right amount of elements 1`] = `
<div
  className="dashboard-container"
>
  <Connect(SideBar) />
  <main>
    <div
      className="welcome-caption"
    >
      <div
        className="mobile-hambuger"
      >
        <a
          className="button-collapse hide-on-large-only "
          data-activates="slide-out"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <span
        className="float-header-with-flex"
      />
      <span>
        <h5
          className="username-format"
        >
          <i
            className="material-icons username-icon"
          >
            account_circle
          </i>
          Welcome, 
        </h5>
      </span>
    </div>
    <div
      className="container"
    >
      <div
        className="edit-wrapper"
      >
        <div>
          <h2>
            Update Profile
          </h2>
        </div>
        <div
          className="account-summary-section"
        >
          <div>
            <form
              className="col s12 update"
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="fullName"
                    name="fullName"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <label
                    className="control-label"
                    htmlFor="fullName"
                  >
                    Fullname :
                  </label>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="userName"
                    name="userName"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <label
                    className="control-label"
                    htmlFor="userName"
                  >
                    Username :
                  </label>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    required={true}
                    type="email"
                    value=""
                  />
                  <label
                    className="control-label"
                    htmlFor="email"
                  >
                    Email :
                  </label>
                </div>
              </div>
              <div
                className="row form-cta"
              >
                <button
                  className="btn shadow-effect"
                  href="#"
                  id="update"
                  type="submit"
                >
                  Update
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`UpdateUser Component snapshot should return user 1`] = `
<div
  className="dashboard-container"
>
  <Connect(SideBar) />
  <main>
    <div
      className="welcome-caption"
    >
      <div
        className="mobile-hambuger"
      >
        <a
          className="button-collapse hide-on-large-only "
          data-activates="slide-out"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <span
        className="float-header-with-flex"
      />
      <span>
        <h5
          className="username-format"
        >
          <i
            className="material-icons username-icon"
          >
            account_circle
          </i>
          Welcome, 
        </h5>
      </span>
    </div>
    <div
      className="container"
    >
      <div
        className="edit-wrapper"
      >
        <div>
          <h2>
            Update Profile
          </h2>
        </div>
        <div
          className="account-summary-section"
        >
          <div>
            <form
              className="col s12 update"
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="fullName"
                    name="fullName"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="Samuel Uyali"
                  />
                  <label
                    className="control-label"
                    htmlFor="fullName"
                  >
                    Fullname :
                  </label>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="userName"
                    name="userName"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="Samuel"
                  />
                  <label
                    className="control-label"
                    htmlFor="userName"
                  >
                    Username :
                  </label>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    required={true}
                    type="email"
                    value="samuel.uyali@gmail.com"
                  />
                  <label
                    className="control-label"
                    htmlFor="email"
                  >
                    Email :
                  </label>
                </div>
              </div>
              <div
                className="row form-cta"
              >
                <button
                  className="btn shadow-effect"
                  href="#"
                  id="update"
                  type="submit"
                >
                  Update
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
